pkg/pmem-csi-operator/controller/deployment/controller_driver.go

```diff
@@ -1065,3 +1100,133 @@ func joinMaps(left, right map[string]string) map[string]string {
 	}
 	return result
 }
+
+// buildFromPreDeployed builds a new Deployment object by fetching details
+// from existing deplyment sub-resources.
+func (d *PmemCSIDriver) buildFromPreDeployed(r *ReconcileDeployment) (*api.Deployment, error) {
```

Comment:
Same here: if an object cannot be found, we shouldn't treat that as an error because otherwise the driver deployment gets stuck forever.

Instead, the operator must recover by using the desired deployment parameters for those objects which don't exist, then later create them.

Needs a test...

Reply:
> Another thought: what happens if the object doesn't exist because the operator we update 
> from didn't create it?

To handle such cases we could change the semantics of Update()/Create() such that it creates if not exists else it updates.

Reply:
> Same here: if an object cannot be found, we shouldn't treat that as an error because otherwise the driver deployment gets stuck forever.
> Instead, the operator must recover by using the desired deployment parameters for those objects which don't exist, then later create them.

I believe this is how the current implementation of this function treats, it doesn't return any error for an object that was not found. Can you please be specific which part of the code does not follow this.